### ISSUE TITLE
A black image was mistakenly generated while running the code in Quick Start

### ISSUE THREAD

**Issue by zc1023** [VectorSpaceLab/OmniGen#21]
I get incorrect results while running the example code.

I install OmniGen via `pip install OmniGen`, and run the example code.

```
from OmniGen import OmniGenPipeline

pipe = OmniGenPipeline.from_pretrained("Shitao/OmniGen-v1")

images = pipe(
    prompt="A curly-haired man in a red shirt is drinking tea.", 
    height=1024, 
    width=1024, 
    guidance_scale=2.5,
    seed=0,
)
images[0].save("example_t2i.png")  # save output PIL Image
```

However, this code generates a black image mistakenly.
![example_t2i](https://github.com/user-attachments/assets/895b5c6a-4ae5-41df-9f25-e26fc8aedc2e)

There is a warming. 
``'
We detected that you are passing `past_key_values` as a tuple of tuples. This is deprecated and will be removed in v4.47. Please convert your cache or use an appropriate `Cache` class (https://huggingface.co/docs/transformers/kv_cache#legacy-cache-format)
``'
And my Python version is `python-3.10`.

**Comment by staoxiao** [comment]
The warning can be ignored. What version of torch are you using?

**Comment by zc1023** [comment]
> The warning can be ignored. What version of torch are you using?

`2.5.0+cu124`

**Comment by Manni1000** [comment]
I once got a black image as a output when i used a uncomon image formart as a input. Maby thats ur problem. Or no u are just generating one without input 🤔

**Comment by VeteranXT** [comment]
Where did you put model file?

**Comment by zc1023** [comment]
> Where did you put model file?
Due to network issues, I downloaded the model from HF and used it locally. 

I have observed that short prompts are more likely to generate black images. Is this due to the padding tokens?

**Comment by VeteranXT** [comment]
C:\Users\Vlada.cache\huggingface\hub\models--Shitao--OmniGen-v1\snapshots\a2fc15e417c06da573546da4578231e8433d05c3

Downloaded by omnigen. When i changed name it started to download that model again.

**Comment by staoxiao** [comment]
@VeteranXT , If you have downloaded all files from HF, you can pass the local path to omnigen: `OmniGenPipeline.from_pretrained("local_path")`

**Comment by zc1023** [comment]
@staoxiao 
I have found the potential cause of generating black images. The [mask of padding tokens](https://github.com/VectorSpaceLab/OmniGen/blob/910a273153cb5aa0b9fe5c2f285e9fd7e5dbc439/OmniGen/pipeline.py#L202) is incorrect. As shown in the following image, the yellow parts in the top rows(padding tokens) should be the causal attention instead of full attention.  So, in some cases, [LLM](https://github.com/VectorSpaceLab/OmniGen/blob/910a273153cb5aa0b9fe5c2f285e9fd7e5dbc439/OmniGen/model.py#L338) crashs,i.e. outputs NULL!
![output](https://github.com/user-attachments/assets/9a6ebaf5-4d07-4c6f-be79-a9c6d5199ec3)

**Comment by USTC-jwn** [comment]
> @staoxiao I have found the potential cause of generating black images. The [mask of padding tokens](https://github.com/VectorSpaceLab/OmniGen/blob/910a273153cb5aa0b9fe5c2f285e9fd7e5dbc439/OmniGen/pipeline.py#L202) is incorrect. As shown in the following image, the yellow parts in the top rows(padding tokens) should be the causal attention instead of full attention. So, in some cases, [LLM](https://github.com/VectorSpaceLab/OmniGen/blob/910a273153cb5aa0b9fe5c2f285e9fd7e5dbc439/OmniGen/model.py#L338) crashs,i.e. outputs NULL! ![output](https://private-user-images.githubusercontent.com/66908299/379759507-9a6ebaf5-4d07-4c6f-be79-a9c6d5199ec3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3NzM0NTEsIm5iZiI6MTcyOTc3MzE1MSwicGF0aCI6Ii82NjkwODI5OS8zNzk3NTk1MDctOWE2ZWJhZjUtNGQwNy00YzZmLWJlNzktYTljNmQ1MTk5ZWMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI0VDEyMzIzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc5M2YxMmEyNmQxMmM4Y2I5YzFjMDlmZjk5YzEyZGI0MmY0OWQxZDZjMGU0MjFkZDA4ODdkNzc4NTEzZDhiYzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.K__bFOPvhkKESjVCvyIXd0YhWnFFvYrJoDR-oZ-CApE)

I have the same problem, how do I solve it to produce normal images？

**Comment by USTC-jwn** [comment]
> pad_mask = torch.tril(torch.ones(size=(pad_l, seq_len)))

With this modification, the output is still a black image. The problem is not solved

**Comment by zc1023** [comment]
>If possible, you could try directly modifying the code at [line 183](https://github.com/VectorSpaceLab/OmniGen/blob/main/OmniGen/processor.py#L188) and see if it works. `pad_mask = torch.ones(size=(pad_l, seq_len))` -> `pad_mask = torch.tril(torch.ones(size=(pad_l, seq_len)))`

don't work. Strangely, it seems related to the Python or package versions, as I can't successfully run your example code. Could you provide the exact package version?

Further, fixing this bug may only require a simple adjustment of the attn-mask of padding tokens.

**Comment by staoxiao** [comment]
@zc1023 I have added a requirments.txt. Besides, someone resolved this issue by updating Torch to a stable version; you can try that as well. I have used Torch 2.3.1 and 2.1.2+cu121 without any problems.

**Comment by kaleyroy** [comment]
> > The warning can be ignored. What version of torch are you using?
> 
> `2.5.0+cu124`

i faced the same issue on ubuntu-20.04-GPU-Geforce_nvidia550.78_cuda12.4

**Comment by kaleyroy** [comment]
> > > The warning can be ignored. What version of torch are you using?
> > 
> > 
> > `2.5.0+cu124`
> 
> i faced the same issue on ubuntu-20.04-GPU-Geforce_nvidia550.78_cuda12.4
> 
> ![image](https://private-user-images.githubusercontent.com/1535437/379997832-a1e748d6-7e2d-442a-a777-d15572a3dba2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4MjU5MzcsIm5iZiI6MTcyOTgyNTYzNywicGF0aCI6Ii8xNTM1NDM3LzM3OTk5NzgzMi1hMWU3NDhkNi03ZTJkLTQ0MmEtYTc3Ny1kMTU1NzJhM2RiYTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMjVUMDMwNzE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjFiMjBlY2MzYWNhNzYwMzE0YzVkNjU3ZjhhOGE1NmEwM2ZiNGViYzNhZWM1NzY5NWYwMjZhM2ZlZWQ3ZTVmZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.r4NuzWjZ6fQXnqXV1GvBqgNe0NdfJjsciE01u87coX4)

#32  it works after update the pytorch version suggested by @escorciav
https://github.com/VectorSpaceLab/OmniGen/issues/32#issuecomment-2434957150

```
pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu124
``

**Comment by kaleyroy** [comment]
> I get incorrect results while running the example code.
> 
> I install OmniGen via `pip install OmniGen`, and run the example code.
> 
> ```
> from OmniGen import OmniGenPipeline
> 
> pipe = OmniGenPipeline.from_pretrained("Shitao/OmniGen-v1")
> 
> images = pipe(
>     prompt="A curly-haired man in a red shirt is drinking tea.", 
>     height=1024, 
>     width=1024, 
>     guidance_scale=2.5,
>     seed=0,
> )
> images[0].save("example_t2i.png")  # save output PIL Image
> ```
> 
> However, this code generates a black image mistakenly. ![example_t2i](https://private-user-images.githubusercontent.com/66908299/379201739-895b5c6a-4ae5-41df-9f25-e26fc8aedc2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4Mjc2ODUsIm5iZiI6MTcyOTgyNzM4NSwicGF0aCI6Ii82NjkwODI5OS8zNzkyMDE3MzktODk1YjVjNmEtNGFlNS00MWRmLTlmMjUtZTI2ZmM4YWVkYzJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI1VDAzMzYyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlMmM5NWMxYjJlNzFmZWNmZWU4MTI2YmVkMGI0MTRiNTQyNjQ4NTA4MGExNGZhZjA0ZmFiNzMxZTQ5Y2RiNzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._VMz-Z2_jSXOvHCFwoCfNR7xwcKEE8k1HVpPNGxq9f0)
> 
> There is a warming. `` ' We detected that you are passing `past_key_values` as a tuple of tuples. This is deprecated and will be removed in v4.47. Please convert your cache or use an appropriate `Cache` class (https://huggingface.co/docs/transformers/kv_cache#legacy-cache-format) ``' And my Python version is `python-3.10`.

you can try this

https://github.com/VectorSpaceLab/OmniGen/issues/32#issuecomment-2434957150

```
pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu124
```

**Comment by zc1023** [comment]
> @zc1023 I have added a requirments.txt. Besides, someone resolved this issue by updating Torch to a stable version; you can try that as well. I have used Torch 2.3.1 and 2.1.2+cu121 without any problems.

The requirements.txt is useful. Thank you! 
The mistake generation is not due to the padding mask. I have changed it to the correct mask, but it didn't work. Why is the padding token id 2 instead of 0? It may lead to unexpected overflows in LLM attention calculation in some cases, right?

Anyway, the requirements are useful! THANK YOU!

**Comment by twoxfh** [comment]
Installing the requirements.txt and lowering the resolution worked for me. It appears the sample code uses slightly more than your typical 24gb gpu. 

The ability to use multiple gpu's would be appreciated :).
